### PR TITLE
Extended format

### PR DESCRIPTION
**Pull request by panpeter**
This PR extends bolds by `__` and `**` and italics by `_` and `*`. It also adds inline code surrounded by ` and code blocks surrounded by ```

### CODE REVIEW

File: src/main/java/com/todoist/markup/MarkupType.java [hunk 1]
```diff
@@ -17,6 +17,16 @@
     ITALIC,
 
     /**
+     * Inline code is surrounded by "`"
```
Comment (goncalossilva) [inline src/main/java/com/todoist/markup/MarkupType.java]: Missing `.` punctuation at the end.

File: src/main/java/com/todoist/markup/MarkupType.java [hunk 2]
```diff
@@ -17,6 +17,16 @@
...
+     */
+    INLINE_CODE,
+
+    /**
+     * Code block is surrounded by "```"
```
Comment (goncalossilva) [inline src/main/java/com/todoist/markup/MarkupType.java]: Missing `.` punctuation at the end.

File: src/main/java/com/todoist/markup/EmojiParser.java [hunk 3]
```diff
@@ -63,12 +63,15 @@ static synchronized void init() throws IOException {
...
 
             // Pattern for standard emojis.
+            patternBuilder.append("(");
             patternBuilder.append(REGEXP_STANDARD_EMOJI);
 
             for(String shortcut : todoistShortcuts.keySet()) {
                 patternBuilder.append("|");
                 patternBuilder.append(Pattern.quote(shortcut));
             }
+            // Exclude links ://
```
Comment (goncalossilva) [inline src/main/java/com/todoist/markup/EmojiParser.java]: Can we improve this comment by stating exactly what happens if we don't do this?

**Comment by goncalossilva** [comment]
Looking good! Left a few comments :smile:

**Comment by panpeter** [comment]
The problem with links is that :// part was converted to 😕/ (:/ was changed to emoji). To solve it I used negative lookahead to check that the pattern is not followed by /.

Actually I just realised that I've used `lookbehind` instead of `lookahead` but the funny thing is that somehow `lookbehind` was working also 😅

**Comment by goncalossilva** [comment]
Ha! The joke's on me — your comment was great but I assumed `://` was just a sad face :joy: 

This leads me to another question: `://` in URLs never have spaces behind or following them. It seems the Web does not allow entries in `emojis_todoist.json` to not be preceded by a space. Why do we allow this? (eg: https://cloudup.com/cmXLHYXjYUv)

**Comment by panpeter** [comment]
Yeah you're right. I've just reused exact pattern from web https://github.com/Doist/ist_libs/blob/beta/python/doist_format/__init__.py
That solves the issue:

screenshot 2016-03-14 22 08 31

**Comment by goncalossilva** [comment]
Even better :+1: Looks great to merge!

**Pull request merged**